core/drivers/geni_uart.c

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: GPL-2.0+
```

Comment:
This will be an issue as it will force the final build to be GPLv2 based, other GPL files are dual licensed with a more permissive license, such as BSD-2 (main one used by optee).

Reply:
The code here and for the clock are both based on U-Boot drivers which are licensed this way, and I don't have permission from the original authors to relicense them.

Reply:
It must have a permissive license or we can't take it.